core/src/main/cljc/jarl/config/schema.cljc

```diff
@@ -0,0 +1,96 @@
+(ns jarl.config.schema
+  (:require [clojure.string  :as str]
+            [jarl.formatting :as fmt]
+            [malli.core      :as m]
+            [malli.error     :as me]
+            [malli.transform :as mt]
+            [malli.util      :as mu]))
+
+(defn fmt-err [s]
+  (fn [{:keys [value]} _]
+    (fmt/sprintf s value)))
+
+(def Credentials
+  [:map
```

Comment:
This map isn't optional, but it can be empty?

Reply:
It's optional on the service object:
```
[:credentials {:optional true} Credentials]]
```
I think that's enough, or WDYT?